### DISCUSSION TITLE
Export state machine to scxml

### DISCUSSION THREAD

**Discussion by viestat** [statelyai/xstate#1665] [Q&A]
I was looking for a way to generate visualisations for my state machines so I can embed them in some documentation pages.

I was wondering if it was possible to export the state machines in **scxml**, so that I can potentially use tools like [state-machine-cat](https://github.com/sverweij/state-machine) (or any tool that takes **scxml** as input) to generate this visualisations from it.

I see there is a package here with that purpose: [`@xstate/scxml`](https://github.com/davidkpiano/xstate/tree/master/packages/xstate-scxml). However it does not seem to be available as a package on npm since it is ignored in the [.changeset/config.json file](https://github.com/davidkpiano/xstate/blob/ff768c7f34a555a6121df4bc69627d6c22d0a7d3/.changeset/config.json#L11-L16).

Please note that while the main question here is on how to export state machines to **scxml**, I would also be happy to be able to use the [`@xstate/viz`](https://github.com/davidkpiano/xstate/tree/master/packages/xstate-viz) package to generate documentation in an automated way.

Thank you so much for the hard work you have put into Xstate!

**Comment by davidkpiano** [comment]
One way you can do this for any graph visualization tool is by converting it to a directed graph first, using [`toDirectedGraph()` from `@xstate/graph`](https://xstate.js.org/docs/packages/xstate-graph/#todirectedgraph-machine).

That returns an object of this structure:

```js
{
  id: '...',
  stateNode: /* StateNode */,
  children: [
    { id: '...', children: [/* ... */], edges: [/* ... */] },
    { id: '...', /* ... */ },
    // ...
  ],
  edges: [
    { source: /* ... */, target: /* ... */, transition: /* ... */ }
    // ...
  ]
}
```

Which makes it easier to convert it to e.g., PlantUML or GraphViz (DOT) formats.
